src/status_im/utils/ethereum/erc20.cljs

```diff
@@ -22,27 +22,99 @@
             [status-im.utils.datetime :as datetime]
             [clojure.string :as string]
             [status-im.utils.security :as security]
+            [status-im.js-dependencies :as dependencies]
             [status-im.utils.types :as types])
   (:refer-clojure :exclude [name symbol]))
 
+(def utils dependencies/web3-utils)
+
+(def abi
+  (clj->js
+   [{:constant        true
+     :inputs          []
+     :name            "name"
+     :outputs         [{:name ""
+                        :type "string"}]
+     :payable         false
+     :stateMutability "view"
+     :type            "function"}
+    {:constant        true
+     :inputs          []
+     :name            "symbol"
+     :outputs         [{:name ""
+                        :type "string"}]
+     :payable         false
+     :stateMutability "view"
+     :type            "function"}
+    {:constant        true
+     :inputs          []
+     :name            "decimals"
+     :outputs         [{:name ""
+                        :type "uint8"}]
+     :payable         false
+     :stateMutability "view"
+     :type            "function"}
+    {:constant        true
+     :inputs          [{:name "_who"
+                        :type "address"}]
+     :name            "balanceOf"
+     :outputs         [{:name ""
+                        :type "uint256"}]
+     :payable         false
+     :stateMutability "view"
+     :type            "function"}
+    {:constant        true
+     :inputs          []
+     :name            "totalSupply"
+     :outputs         [{:name ""
+                        :type "uint256"}],
+     :payable         false
+     :stateMutability "view"
+     :type            "function"}
+    {:constant        false
+     :inputs          [{:name "_to"
+                        :type "address"}
+                       {:name "_value"
+                        :type "uint256"}]
+     :name            "transfer"
+     :outputs         [{:name ""
+                        :type "bool"}],
+     :payable         false
+     :stateMutability "nonpayable"
+     :type            "function"}
+    {:anonymous false
+     :inputs    [{:indexed true
+                  :name    "from"
+                  :type    "address"},
+                 {:indexed true
+                  :name    "to"
+                  :type    "address"},
+                 {:indexed false
+                  :name    "value"
+                  :type    "uint256"}]
+     :name      "Transfer"
+     :type      "event"}]))
+
+(defn get-instance* [web3 contract]
+  (.at (.contract (.-eth web3) abi) contract))
+
+(def get-instance
+  (memoize get-instance*))
+
 (defn name [web3 contract cb]
-  (ethereum/call web3 (ethereum/call-params contract "name()") cb))
+  (.name (get-instance web3 contract) cb))
```

Comment:
could you elaborate on this decision?

Reply:
Yes. The previous solution returned a hex string that had to be parsed manually. That would have involved too much boilerplate and it looked unstable - the same string would be decoded fine in In the new one `web3` does it automatically.
If that's the decision you asked about.

Reply:
I thought @yenda and @jeluard  introduced `ethereum/call-params` to replace web3 usage, and we want to get rid of web3

Reply:
@yenda @jeluard Any context guys?

Reply:
in that case it's still using the old ways call-params doesn't really replace web3 usage, @flexsurfer is probably refering to abi-spec/encode and abi-spec/decode functions

Reply:
Ok so we're good to go then?

Reply:
well you can replace the current thing by abi-spec/encode and abi-spec decode

Reply:
that didn't work, I'd get garbled text after decode.